service/worker/scanner/scanner_test.go

```diff
@@ -0,0 +1,216 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package scanner
+
+import (
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/suite"
+
+	"go.temporal.io/server/api/historyservicemock/v1"
+	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+	p "go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/common/sdk"
+	"go.temporal.io/server/common/testing/mocksdk"
+)
+
+type scannerTestSuite struct {
+	suite.Suite
+}
+
+func TestScanner(t *testing.T) {
+	suite.Run(t, new(scannerTestSuite))
+}
+
+func (s *scannerTestSuite) TestScannerEnabled() {
+	defer func(originalDelay time.Duration) {
+		scannerStartUpDelay = originalDelay
+	}(scannerStartUpDelay)
```

Comment:
will this work after you set the delay to 0?

Reply:
Yes. All of the deferred function's arguments are computed immediately. For example:

```
package main

func main() {
	x := 0
	defer func(x int) {
		println(x) // prints 0
	}(x)
	x = 42
}
```